internal/proxy/reverseproxy/reverseproxy.go

```diff
@@ -50,4 +68,50 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		logger.Warnf("Error %q - code: %d", rr.err, rr.statusCode)
 		http.Redirect(rw, req, rp.RedirectURL.String(), http.StatusFound)
 	}
+
+	if rr.statusCode == http.StatusForbidden {
+		var err error
+		rp.IsValidSession, err = checkSessionValidity(req)
+		if err != nil {
+			logger.Error(err)
+		}
+		http.Redirect(rw, req, rp.RedirectURL.String(), http.StatusFound)
+	}
+
+}
+
+func checkSessionValidity(req *http.Request) (bool, error) {
+	requestURL := req.URL
+	cookies := req.Cookies()
+
+	jenkinsURL := fmt.Sprintf("%s://%s", requestURL.Scheme, requestURL.Host)
+
+	for _, cookie := range cookies {
+		if cookiesutil.IsSessionCookie(cookie) {
+
+			req, _ := http.NewRequest("GET", jenkinsURL, nil)
+			req.AddCookie(cookie)
+
+			c := http.DefaultClient
+			resp, err := c.Do(req)
+			if err != nil {
+				return false, err
+			}
+			defer resp.Body.Close()
+
+			switch resp.StatusCode {
+			case http.StatusOK:
+				return true, err
+			case http.StatusUnauthorized:
+				return false, err
```

Comment:
I should get rid off this unnecessary status check, since we know that the status returned would be Forbidden if not logged in.